source/base/ByteBuffer.ooc

```diff
@@ -123,7 +123,7 @@ _RecyclableByteBuffer: class extends ByteBuffer {
 		bin := This _getBin(size)
 		This _lock lock()
 		for (i in 0 .. bin count)
-			if ((bin[i] size) == size) {
+			if ((bin[i] size) == size && (bin[i] _allocator == allocator)) {
```

Comment:
Oh, that was easy. While you're here, there are two not-needed pairs of parentheses.